cabal-testsuite/PackageTests/Help/HelpPrintsConfigFile/help-prints-config-file.test.hs

```diff
@@ -1,25 +1,50 @@
--- Andreas Abel, 2024-01-13
+-- Andreas Abel, 2024-01-13, 2024-01-28
 --
--- Ensure that the last line of the help text is the name of the config file.
+-- Ensure that the help text prints the name of the config file, following a fixed text.
 -- This invariant is used by clients such as the Haskell setup github action.
 -- See: https://github.com/haskell-actions/setup/pull/63
 
+-- The end of the help text should look like:
+--
+-- > You can edit the cabal configuration file to set defaults:
+-- >   <<HOME>>/.cabal/config
+-- > This file will be generated with sensible defaults if you run 'cabal update'.
+--
+-- The last line is only present when the file does *not* exist.
+--
+-- So while usually the last line is the name of the config file,
+-- the correct way is to take the line after the fixed text "You can edit...".
+
 import Distribution.Utils.String (trim)
+import System.Directory (removeFile)
 import Test.Cabal.Prelude
 
 main = cabalTest $ do
   env <- getTestEnv
-  res <- cabal' "--help" []
+  let configFile = testUserCabalConfigFile env
 
-  -- The end of the help text should be something like:
-  --
-  -- > You can edit the cabal configuration file to set defaults:
-  -- >   <<HOME>>/.cabal/config
-  --
-  -- So trimming the last line will give us the name of the config file.
-  let configFile = trim . last . lines . resultOutput $ res
+  -- Test 1: with config file present.
+  test configFile "(config file exists)"
 
-  -- Verify that this is indeed the config file.
-  assertEqual "Last line of help text should be name of the config file"
-    (testUserCabalConfigFile env)
+  -- Test 2: with config file absent.
+  liftIO $ removeFile configFile
+  test configFile "(config file deleted)"
+
+test configFile info = do
+  res <- cabal' "--help" []
+  assertEqual (unwords ["Help text contains name of the config file after the marker", info])
```

Comment:
```suggestion
  assertEqual ("Help text contains name of the config file after the marker \"" ++ info ++ "\"")
```

Reply:
Maybe a misunderstanding.  `info` isn't the text of the marker, but some additional info about the test (whether we testing with present or absent config file).  So, no need to quote the `info`.